src/Widgets/CameraView.vala

```diff
@@ -205,39 +221,56 @@ public class Camera.Widgets.CameraView : Gtk.Stack {
                 "videoscale name=videoscale"
             );
 
-            pipeline.add (device_src);
-            device_src.link (pipeline.get_by_name ("decodebin"));
-            tee = pipeline.get_by_name ("tee");
-            hflip = (pipeline.get_by_name ("hflip") as Gst.Video.Direction);
-            color_balance = (pipeline.get_by_name ("balance") as Gst.Video.ColorBalance);
+            preview_pipeline.add (device_src);
+            dynamic Gst.Element capsfilter = preview_pipeline.get_by_name ("capsfilter");
+            device_src.link (capsfilter);
+
+            // Ensure action state changes with caps filter
+            capsfilter.get_static_pad ("src").add_probe (Gst.PadProbeType.EVENT_BOTH, (pad, info) => {
+                unowned Gst.Event? event = info.get_event ();
+                if (event.type == Gst.EventType.CAPS) {
+                    //TODO Find correct resolution index and update action state.
```

Comment:
I have inserted a placeholder here as the existing code did not seem to be working and I am not sure under what circumstances this event would get triggered other than through user interaction with the app (which is already catered for).